provider/datagen/src/transform/cldr/units/mod.rs

```diff
@@ -2,34 +2,137 @@
+            };
+
+            constants_map_in_str_form.insert(cons_name, (num, den, constant_type));
         }
 
-        let result = UnitsConstantsV1 {
-            constants_map: ZeroMap::from_iter(constants_map),
-        };
+        // This loop iterates over the constants, replacing any string values with their corresponding constant values.
+        let mut num_of_const_with_text;
+        loop {
+            num_of_const_with_text = 0;
+            let mut constants_with_constants_map_replaceable =
+                BTreeMap::<&str, (Vec<String>, Vec<String>, ConstantType)>::new();
+            for (cons_name, (num, den, constant_type)) in constants_map_in_str_form.iter() {
+                let mut temp_num = num.clone();
+                let mut temp_den = den.clone();
+                let mut temp_constant_type = *constant_type;
+
+                for i in 0..temp_num.len() {
+                    if !contains_alphabetic_chars(temp_num[i].as_str())
+                        || is_scientific_number(temp_num[i].as_str())
+                    {
+                        continue;
+                    }
+
+                    num_of_const_with_text += 1;
+                    if let Some((rnum, rden, rconstant_type)) =
+                        constants_map_in_str_form.get(temp_num[i].as_str())
+                    {
+                        temp_num.remove(i);
+                        // append the elements in rnum to num and rden to den
+                        temp_num.append(&mut rnum.clone());
+                        temp_den.append(&mut rden.clone());
+
+                        if *rconstant_type == ConstantType::Approximate {
+                            temp_constant_type = ConstantType::Approximate;
+                        }
+                    }
+                }
+
+                for i in 0..temp_den.len() {
+                    if !contains_alphabetic_chars(temp_den[i].as_str())
+                        || is_scientific_number(temp_den[i].as_str())
+                    {
+                        continue;
+                    }
+
+                    num_of_const_with_text += 1;
+                    if let Some((rnum, rden, rconstant_type)) =
+                        constants_map_in_str_form.get(temp_den[i].as_str())
+                    {
+                        temp_den.remove(i);
+                        // append the elements in rnum to den and rden to num
+                        temp_num.append(&mut rden.clone());
+                        temp_den.append(&mut rnum.clone());
+
+                        if *rconstant_type == ConstantType::Approximate {
+                            temp_constant_type = ConstantType::Approximate;
+                        }
+                    }
+                }
+
+                constants_with_constants_map_replaceable
+                    .insert(cons_name, (temp_num, temp_den, temp_constant_type));
+            }
+
+            constants_map_in_str_form.clear();
```

Comment:
You don't need to clear the map because you're dropping it in the next line.

```suggestion
```

Reply:
done